dvc/stage/run.py

```diff
@@ -46,13 +48,14 @@ def warn_if_fish(executable):
 @unlocked_repo
 def cmd_run(stage, *args, checkpoint_func=None, **kwargs):
     kwargs = {"cwd": stage.wdir, "env": fix_env(None), "close_fds": True}
+    cmd = stage.cmd if isinstance(stage.cmd, list) else [stage.cmd]
```

Comment:
The issue you linked to is also about multiline string.
Right now, it's not officially supported, but works depending on the shell.
What do you think of it? Should we support it?

Reply:
I am not sure about your question: the issue mentions:
 - piping: this shout stay as is as stated by @efiop 
 - `&&`: the purpose of this pr: officially allowing the user to define sequential commands in a single stage from its yaml definition.

I don't see much space to think about something else

Reply:
@ClementWalter, as you are proposing to close the issue related to multiline/multiple commands, I just wanted to know what your opinion is on that, and if we should support it, as this list and the multiline string seems to be allowing the user the same thing.

Reply:
I think I don't understand what you are trying to ask me

Reply:
ok you mean @skshetry if all this work is worth doing because eventually it is only syntactic sugar for yaml?

Reply:
ok I think I got it, just re-reading from the beginning. You mean: should we also support the parallel case, ie multi line string where every line should a command that runs in parallel?

Reply:
@ClementwWalter I believe @skshetry was wondering about your opinion on whether or not we should treat multiline string as a list of commands when running it. https://github.com/iterative/dvc/issues/4373#issuecomment-728359883 I've mentioned that it should probably stay as it is, but there is an idea that maybe we should handle that better, similar as gha yaml does it. Let's keep that issue opened for that last question.

Reply:
@efiop yes! actually re-reading this morning and re-reading yaml parsing rules it became 100% clear to me what was @skshetry's point, sorry about that yesterday. My guess is that we should not try to parse user input. Maybe we could add a flag in the stage definition about the desired behavior with a list of command: stop on fail or keep going?